### ISSUE TITLE
WSL command line compatability (Creators Update)

### ISSUE THREAD

**Issue by ryanerwin** [atom/atom#14258]
### Prerequisites

* [X] Put an X between the brackets on this line if you have done all of the following:
    * Reproduced the problem in Safe Mode: 
    * Followed all applicable steps in the debugging guide: 
    * Checked the FAQs on the message board for common solutions:
    * Checked that your issue isn't already filed: 
    * Checked that there is not already an Atom package that provides the described functionality: 

### Description

Now that [WSL/Windows Interop is officially released](https://msdn.microsoft.com/en-us/commandline/wsl/interop), would be good to update the Windows Atom wrapper scripts. 

When running "atom" or "apm" inside bash.exe (Windows Creators Update Build 15063.138) the WSL environment isn't detected by the scripts inside %UserProfile%\AppData\Local\atom\bin

WSL's bash.exe environment will call the shell script (as opposed to batch files) by default

### Steps to Reproduce

1. Open the WSL environment (<kbd>Win</kbd> + <kbd>R</kbd> + <kbd>`bash.exe`</kbd> + <kbd>Enter</kbd>)
2. Inside WSL environment, open Atom (<kbd>`atom.exe`</kbd>+<kbd>Enter</kbd> 

**Expected behavior:** Expected atom and/or apm to work normally

**Actual behavior:** 

**ATOM**
```
/mnt/c/Users/ryan/AppData/Local/atom/bin/../app-1.16.0/resources/cli/atom.sh: line 9: pwd: -W: invalid option
pwd: usage: pwd [-LP]
'/atom.cmd' is not recognized as an internal or external command,
operable program or batch file.
```

**APM**

```
/mnt/c/Users/ryan/AppData/Local/atom/bin/../app-1.16.0/resources/cli/../app/apm/bin/apm: line 37: /mnt/c/Users/ryan/AppData/Local/atom/app-1.16.0/resources/app/apm/bin/node: No such file or directory
```

**Reproduces how often:** Every time

### Versions

```
 % **atom --version**
Atom    : 1.16.0
Electron: 1.3.13
Chrome  : 52.0.2743.82
Node    : 6.5.0

 % **apm --version**
Microsoft Windows [Version 10.0.15063]
apm  1.16.1
npm  3.10.5
node 6.9.5 x64
python
git 2.10.2.windows.1
visual studio 2015
```

### Additional Information

For now, it's easy enough to work around this issue by creating an alias in your WSL shell, for example:

```
alias atom='cmd.exe /C C:\\Users\\ExampleUser\\AppData\\Local\\atom\\bin\\atom.cmd'
alias apm='cmd.exe /C C:\\Users\\ExampleUser\\AppData\\Local\\atom\\bin\\apm.cmd'
```

Ideally we could rewrite the wrapper files for the Windows package of Atom to work seamlessly whether running from the Start Menu, cmd.exe, bash.exe, or any of the other Cygwin derived environments. Over the near term, I expect that WSL will replace all of the Cygwin derived environments, greatly simplifying most users development environments.

**Comment by rsese** [comment]
Hmmm, maybe a dupe/at least related to https://github.com/atom/atom/issues/12519?

@ryanerwin, do you think Damien's comment here addresses what you're asking for in this issue?
 /cc https://github.com/atom/atom/issues/12519#issuecomment-293121265

**Comment by ryanerwin** [comment]
I saw issue #12519 but this is not related.

In #12519 @bergerd1 is hoping to execute the Atom binary inside of the Windows Subsystem for Linux (ie, he's looking to run the ELF binaries)

however @damieng replied that:
> The top request right now on WSL is the ability to launch Windows applications from the Bash command line which would be our preferred way of using Atom from the WSL.
> The amount of effort to get the Linux version running in WSL would be non-trivial given the early state of WSL and isn't something the core team could commit to at this time.

The Windows/Bash interop that was requested has been delivered in the creators update - ([documentation here](https://msdn.microsoft.com/en-us/commandline/wsl/interop)). I'm just suggesting an update of the shell scripts that wrap around the Windows (EXE) version of Atom. Currently these shell scripts detect support running under cmd.exe or under Cygwin. I'm just suggesting updating the wrapper scripts to work under WSL, which should soon replace Cygwin/Msys/Bash-Git/etc simplifying Web Development environments on Windows.

Specifically, updating [atom.sh ](https://github.com/atom/atom/blob/master/resources/win/atom.sh) and the `apm` wrappers to support WSL with a check along the lines of:

```
if [ grep -q Microsoft /proc/sys/kernel/osrelease ]; then
```

There isn't yet a WSL binary that I've found along the lines of Cygwin's cygpath ([source code here](https://cygwin.com/git/gitweb.cgi?p=newlib-cygwin.git;a=blob;f=winsup/utils/cygpath.cc;h=ccae98f77bf5b2e75df14e7f25429e8fd7ad540a;hb=HEAD)) for Linux to Windows path translation, but it could be rolled easily enough in a shell script.

There is a small [cygpath like wrapper posted on reddit](https://www.reddit.com/r/bashonubuntuonwindows/comments/642dru/i_made_a_wrapper_for_running_ubuntu_programs_on/), but not yet as polished as cygpath.

**Comment by damieng** [comment]
If we want to just do it for /mnt/drive we could easily do that at script level for now.  I'm not sure anything outside of /mnt/ maps to Windows.

**Comment by damieng** [comment]
@ryanerwin Can you try the atom.sh from https://github.com/atom/atom/blob/b5c4493f5776dad0705dd6df31d562287447ac4d/resources/win/atom.sh - put this alongside your existing atom.sh in `%appdata%\atom\resources\cli` - perhaps as atom2.sh and let me know if this works for you.

**Comment by ryanerwin** [comment]
You are correct @damieng, at this time nothing outside of **/mnt** maps to windows by default.
According to [Bash on Windows Ticket 1524](https://github.com/Microsoft/BashOnWindows/issues/1524) it's feasible to export your WSL root / via WebDAV, and map that to a drive letter, but this doesn't seem like an ideal solution.

With the Creators Update, Microsoft makes %UserProfile% (Your _Windows Home Folder_) the default path when you open your terminal, so I think the WSL teams vision is that we use /mnt for accessing all of the files that we're actually working on, and use the Linux files as scaffolding for our projects... Personally, I think I'll setup the WebDAV mapped drive so that I can use atom across the file system.

**Tested your updated version of [atom.sh](https://github.com/atom/atom/blob/b5c4493f5776dad0705dd6df31d562287447ac4d/resources/win/atom.sh) and that works perfectly. Thanks!** 

Need a similar tweak to the shell scripts that wrap around **apm** and the Atom bundled **npm**.

**Comment by rsese** [comment]
Thanks for clarifying for @ryanerwin and thanks for the 👓  @damieng :bow:

**Comment by damieng** [comment]
Merged into apm and atom master branches - should make it in to 1.18 (no eta on that yet)

**Comment by ryanerwin** [comment]
@damieng Do you want to make a similar update to cli/apm.sh ?

Without doing an update, I get:

```
 % apm
/mnt/c/Users/ryan/AppData/Local/atom/bin/../app-1.16.0/resources/cli/../app/apm/bin/apm: line 37: /mnt/c/Users/ryan/AppData/Local/atom/app-1.16.0/resources/app/apm/bin/node: No such file or directory
```

**Comment by damieng** [comment]
cli/apm.sh is a stub that calls apm from the apm package. That was updated as part of this but won't be available until 1.18 ships.

If you want to try it out you can copy https://github.com/atom/apm/blob/master/bin/apm over your apm file in %appdata%\Atom\app-1.16.0\resources\app\apm\bin\apm file
